core/src/main/scala/zio/logging/Logger.scala

```diff
@@ -0,0 +1,64 @@
+package zio.logging
+
+import zio.clock._
+import zio.console._
+import zio.internal.Tracing
+import zio.internal.stacktracer.Tracer
+import zio.internal.stacktracer.ZTraceElement.{ NoLocation, SourceLocation }
+import zio.internal.stacktracer.impl.AkkaLineNumbersTracer
+import zio.internal.tracing.TracingConfig
+import zio.{ FiberRef, UIO, ZIO }
+
+/**
+ * A logger of strings.
+ */
+trait Logger extends LoggerLike[String]
+object Logger {
+
+  private val tracing = Tracing(Tracer.globallyCached(new AkkaLineNumbersTracer), TracingConfig.enabled)
+
+  private[logging] def classNameForLambda(lambda: => AnyRef) =
```

Comment:
This will be a very slow way to get the enclosing class of the lambda. I'm not saying _don't_ do it (maybe it is useful enough, I am not sure), but we have enough work cut out for us if we want to make this fast as it is.

Reply:
I added this as an option, sometimes useful but I have mix feeling about this as well. it's used only when ```LogAnnotation.Name``` is empty.

Reply:
let me know if you want to keep it.

Reply:
I think I would prefer _not_ to keep it unless users ask for it... they can always do this themselves with a thin wrapper.

Reply:
sure I will move this to slf4j and hide it.